### PR TITLE
Add models for client and server according to spec

### PR DESCRIPTION
**Pull request by rsachdevch**
Resolves FAML-332

### CODE REVIEW

**Comment by rsachdevch** [comment]
> Should the package name be pscdiscrepanciesapi?

Sure, I'll rename the base package to that

**Comment by ch-tstrickland** [comment]
Looking further into it, I see that extensions called the end of their project's package `extensions.api`, rather than `extensionsapi`. I don't mind which we choose, but we should probably establish what the convention is.
Also, the lack of unit tests seems to be upsetting sonar. Not sure what to do about that.

**Comment by hwilliams1** [comment]
Steve was saying that if the models are moved to a model directory (see https://github.com/companieshouse/company-accounts.web.ch.gov.uk for an example) then the models are automatically excluded from Sonar so could be better to do that?

**Comment by rsachdevch** [comment]
> Steve was saying that if the models are moved to a model directory (see https://github.com/companieshouse/company-accounts.web.ch.gov.uk for an example) then the models are automatically excluded from Sonar so could be better to do that?

Will change the package name

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/model/PscDiscrepancyReportEntity.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package uk.gov.ch.pscdiscrepanciesapi.model;
+
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+import java.time.LocalDateTime;
+
+@Document(collection = "psc_discrepancy_reports")
+class PscDiscrepancyReportEntity {
+    @Field("_id")
```
Comment (sroberts2) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/model/PscDiscrepancyReportEntity.java]: @Id to label this as the id for the table.

**Pull request merged**